pkg/loki/labels.go

```diff
@@ -20,6 +20,10 @@ func GetLokiLabels(desired *flowslatest.FlowCollectorSpec) []string {
 		indexFields = append(indexFields, constants.ClusterNameLabelName)
 	}
 
+	if desired.Processor.AddZone != nil && *desired.Processor.AddZone {
```

Comment:
after rebasing you could use that:
```suggestion
	if helper.IsZoneEnabled(&desired.Processor) {
```